sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetReadState.java

```diff
@@ -131,9 +157,10 @@ long currentRangeEnd() {
    * Advance the current offset and rowId to the new values.
    */
   void advanceOffsetAndRowId(int newOffset, long newRowId) {
-    valuesToReadInBatch -= (newOffset - offset);
+    rowsToReadInBatch -= (newOffset - levelOffset);
     valuesToReadInPage -= (newRowId - rowId);
-    offset = newOffset;
+    levelOffset = newOffset;
+    valueOffset = newOffset;
```

Comment:
So we always advance same range to level and value offsets? But I think they are not always the same, right?

Reply:
This method is only called for reading non-repeated values in which case they are always the same. For the other case it is not through this method.